x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTrunc.java

```diff
@@ -127,9 +234,44 @@ public static Part resolve(String truncateTo) {
             return DateTimeField.resolveMatch(NAME_TO_PART, truncateTo);
         }
 
-        public ZonedDateTime truncate(ZonedDateTime dateTime) {
+        public Object truncate(Object dateTime) {
             return truncateFunction.apply(dateTime);
         }
+
+        private static Object truncateIntervalSmallerThanWeek(Object r, ChronoUnit unit){
+            if (r instanceof IntervalDayTime) {
+                Duration d = ((IntervalDayTime) r).interval();
+                int isNegative = 1;
+                if (d.isNegative()) {
+                    d = d.negated();
+                    isNegative = -1;
+                }
+                long durationInSec = d.getSeconds();
+                long day = durationInSec / SECONDS_PER_DAY;
+                System.out.println(day);
```

Comment:
Please remove.

Reply:
I used too much debugger but for one test cases i just tried to use it and i forgot to remove :D.

Reply:
No worries, happens to everyone.